app/src/main/java/com/example/newsapp/data/repository/EmailPreference.kt

```diff
@@ -1,25 +1,35 @@
 package com.example.newsapp.data.repository
 
 import android.content.Context
+import android.content.SharedPreferences
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
 import com.example.newsapp.util.Constants
 
 class EmailPreference(context: Context) {
 
-    val preference = context.getSharedPreferences(Constants.shared_preferences_name, Context.MODE_PRIVATE)
+    val masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)
+    val preference: SharedPreferences = EncryptedSharedPreferences.create(
```

Comment:
@jnandrawos no need for casting to SharedPreferences, it automatically takes the class type if set as val